weather_dl/download_pipeline/manifest.py

```diff
@@ -242,22 +436,121 @@ def _update(self, download_status: DownloadStatus) -> None:
                 logger.debug(download_status)
 
 
+class BQManifest(Manifest):
+    """Writes a JSON representation of the manifest to BQ file.
+
+    This is an append-only implementation, the latest value in the manifest
+    represents the current state of a download.
+    """
+
+    _lock = threading.Lock()
+
+    def __init__(self, location: Location) -> None:
+        super().__init__(Location(location[5:]))
+        TABLE_SCHEMA = [
+            bigquery.SchemaField('selection', 'JSON', mode='REQUIRED'),
+            bigquery.SchemaField('location', 'STRING', mode='REQUIRED'),
+            bigquery.SchemaField('stage', 'STRING', mode='NULLABLE'),
+            bigquery.SchemaField('status', 'STRING', mode='REQUIRED'),
+            bigquery.SchemaField('error', 'STRING', mode='NULLABLE'),
+            bigquery.SchemaField('user', 'STRING', mode='REQUIRED'),
+            bigquery.SchemaField('size', 'FLOAT', mode='NULLABLE'),
+            bigquery.SchemaField('scheduled_time', 'TIMESTAMP', mode='NULLABLE'),
+            bigquery.SchemaField('retrieve_start_time', 'TIMESTAMP', mode='NULLABLE'),
+            bigquery.SchemaField('retrieve_end_time', 'TIMESTAMP', mode='NULLABLE'),
+            bigquery.SchemaField('fetch_start_time', 'TIMESTAMP', mode='NULLABLE'),
+            bigquery.SchemaField('fetch_end_time', 'TIMESTAMP', mode='NULLABLE'),
+            bigquery.SchemaField('download_start_time', 'TIMESTAMP', mode='NULLABLE'),
+            bigquery.SchemaField('download_end_time', 'TIMESTAMP', mode='NULLABLE'),
+            bigquery.SchemaField('upload_start_time', 'TIMESTAMP', mode='NULLABLE'),
+            bigquery.SchemaField('upload_end_time', 'TIMESTAMP', mode='NULLABLE'),
```

Comment:
Let's add descriptions for each of these fields. e.g. users should have an answer to the question: "are these sizes in bytes or GB?" A description would make it easy to know.

Reply:
Done.